lib/components/narrative/line-itin/connected-itinerary-body.js

```diff
@@ -61,6 +62,30 @@ class ConnectedItineraryBody extends Component {
     } = this.props
     const { LegIcon } = this.context
 
+    let [pickupAdded, dropOffAdded] = [false, false]
+
+    // Support OTP1 flex messages in Trip Details
+    itinerary.legs = itinerary.legs.map((oldLeg) => {
+      const leg = clone(oldLeg)
+      if (!!leg.boardRule && !leg.pickupBookingInfo && !pickupAdded) {
+        pickupAdded = true
+        leg.pickupBookingInfo = {
```

Comment:
Do we want this applied to each leg or only the first leg that satisfies the if condition above?

Reply:
Then maybe `clonedItinerary.legs.find` will work better than the `if` and the `let` assignments.

Reply:
Would the best place to do that itinerary "transformation" be in the first redux `connect` statement that extracts the itinerary from OTP itinerary results.

Reply:
I think we decided to do it here since it's a "transformation" that's very specific to the components called in this file. This is a bit of a hack, so I thought it makes sense to have these weird extra fields permeate as few points of the code as possible

Reply:
I think we need to use `.map` since we're updating the leg's `leg.pickupBookingInfo`. If using `.find` here we'd need to update via an index

Reply:
Hmm I disagree, if you use `find` you should be able to update that leg directly. In doing so you won't need the `let` variables you defined above.
```js
const legToUpdate = clonedItinerary.legs.find(<condition>)
if (legToUpdate) {
  // Should update the applicable leg object in clonedItinerary.legs
  legToUpdate.pickupBookingInfo = { ... }
}
```